src/viur/core/render/html/env/viur.py

```diff
@@ -150,63 +156,66 @@ def getSkel(render: Render, module: str, key: str = None, skel: str = "viewSkel"
+    :param skel_arg: Optional skeleton arguments to be passed to the skel-function (e.g. for Tree-Modules)
 
     :returns: dict on success, False on error.
     """
-    if module not in dir(conf.main_app):
-        logging.error(f"getSkel called with unknown {module=}!")
-        return False
+    if not (obj := getattr(conf.main_app, module, None)):
+        raise ValueError(f"getSkel: Can't read a skeleton from unknown module {module!r}")
 
-    obj = getattr(conf.main_app, module)
+    if not getattr(obj, "html", False):
+        raise PermissionError(f"getSkel: module {module!r} is not allowed to be accessed")
 
-    if skel in dir(obj):
-        skel = getattr(obj, skel)()
+    # Retrieve a skeleton
+    skel = getattr(obj, skel)(*skel_args)
+    if not isinstance(skel, SkeletonInstance):
+        raise RuntimeError("getSkel: Invalid skel name provided")
 
-        if isinstance(obj, prototypes.singleton.Singleton) and not key:
-            # We fetching the entry from a singleton - No key needed
-            key = db.Key(skel.kindName, obj.getKey())
-        elif not key:
-            logging.info("getSkel called without a valid key")
-            return False
-
-        if not isinstance(skel, SkeletonInstance):
-            return False
-
-        if "canView" in dir(obj):
-            if not skel.fromDB(key):
-                logging.info(f"getSkel: Entry {key} not found")
-                return None
-            if isinstance(obj, prototypes.singleton.Singleton):
-                isAllowed = obj.canView()
-            elif isinstance(obj, prototypes.tree.Tree):
-                k = db.Key(key)
-                if k.kind.endswith("_rootNode"):
-                    isAllowed = obj.canView("node", skel)
-                else:
-                    isAllowed = obj.canView("leaf", skel)
-            else:  # List and Hierarchies
-                isAllowed = obj.canView(skel)
-            if not isAllowed:
-                logging.error(f"getSkel: Access to {key} denied from canView")
-                return None
-        elif "listFilter" in dir(obj):
-            qry = skel.all().mergeExternalFilter({"key": str(key)})
-            qry = obj.listFilter(qry)
-            if not qry:
-                logging.info("listFilter permits getting entry, returning None")
-                return None
-
-            skel = qry.getSkel()
-            if not skel:
-                return None
-
-        else:  # No Access-Test for this module
-            if not skel.fromDB(key):
-                return None
-        skel.renderPreparation = render.renderBoneValue
-        return skel
+    if isinstance(obj, prototypes.singleton.Singleton) and not key:
+        # We fetching the entry from a singleton - No key needed
+        key = db.Key(skel.kindName, obj.getKey())
 
-    return False
+    elif not key:
+        raise ValueError("getSkel has to be called with a valid key!")
+
+    if "canView" in dir(obj):
```

Comment:
This is more performant, I think.

```suggestion
    if hasattr(obj, "canView"):
```